src/plugins/data/public/resources/resource_client_factory.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { HttpSetup } from 'opensearch-dashboards/public';
+import { PrometheusResourceClient } from './prometheus_resource_client';
+
+type DataConnectionType = 'prometheus';
+
+export class ResourceClientFactory {
```

Comment:
ik just feature branch but would this be essentially be accomplished with a route on the server side

Reply:
Yeah, this hits the `data_enhancements` resource API route in OSD server which proxies the request to the sql plugin in OS.